### DISCUSSION TITLE
Is there a way to wait specific websocket event data?

### DISCUSSION THREAD

**Discussion by agugut-nexgen** [cypress-io/cypress#15821] [Questions and Help]
Hello, Is there a way to wait until a websocket connection get specific event data?
I have the following code that start a wss
```
const webSocket = new WebSocket(connectionUrl)
```
Then I try to do
```
 webSocket.onmessage = function (event) {
                    if ((event.data as string).toLowerCase().includes('sometext')) {
                                customerWebSocket.close();
                            return true;
                        }
                    }
                };
```
But I don't find a way to wait specific event.data
I trying to use https://github.com/lensesio/cypress-websocket-testing but this plugin didn't work
Somebody can help me?

**Comment by suppadeliux** [comment]
I was thinking about doing the same thing, because my tests depend on the state from my data in websocket event, and not the data of http requests!. Sad that this has not been answered since april 2021!

**Comment by cipagauta-david** [comment]
The same here, any idea?
